02-express-tutorial/app.js

```diff
@@ -1,60 +1,137 @@
-
-//middleware for stattic files from 'public' directory
-app.use(express.static("./public"));
-
-//routes for /api/v1/test to return JSON
-app.get("/api/v1/test", (req, res) => {
-    res.json({ message: "It worked!" });
-  });
-
-//routes to return all products
-app.get("/api/v1/products", (req, res) => {
-    res.json(products);
-  });
-
-//get product by ID
-app.get("/api/v1/products/:productID", (req, res) => {
-    const idToFind = parseInt(req.params.productID);
-    const product = products.find((p) => p.id === idToFind);
-  
-    if (!product) {
-      return res.status(404).json({ message: "That product was not found." });
-    }
-  
-    res.json(product);
-  });
-
-// query products
-app.get("/api/v1/query", (req, res) => {
-    const { search, limit } = req.query;
-    let filteredProducts = products;
-  
-    if (search) {
-      filteredProducts = filteredProducts.filter((product) =>
-        product.name.toLowerCase().startsWith(search.toLowerCase())
-      );
-    }
-  
-    if (limit) {
-      filteredProducts = filteredProducts.slice(0, parseInt(limit));
+
+const express = require('express')
+const app = express()
+const logger = require('./logger')
+const { products, people } = require("./data");
+const peopleRouter = require("./routes/people");
+const cookieParser = require("cookie-parser");
+
+//apply logger for single route
+app.get('/', logger, (req, res) => {
+  res.send('Home')
+})
+app.get('/about', logger, (req, res) => {
+  res.send('About')
+})
+app.get('/api/products', logger, (req, res) => {
+  res.send('Products')
+})
+
+//apply logger globally
+app.use(logger)
+app.get('/',  (req, res) => {
+  res.send('Home')
+})
+app.get('/about', (req, res) => {
+  res.send('About')
+})
+app.get('/api/products', (req, res) => {
+  res.send('Products')
+})
+
+//GET /api/v1/people request:
+app.get('/api/v1/people', (req, res) => {
+  res.json(people);
+});
+
+//POST request 
+// static assets
+app.use(express.static('./methods-public'))
+// parse form data
+app.use(express.urlencoded({ extended: false }))
+// parse json
+app.use(express.json())
+
+app.get('/api/people', (req, res) => {
+  res.status(200).json({ success: true, data: people })
+})
+
+app.post('/api/people', (req, res) => {
+  const { name } = req.body
+  if (!name) {
+    return res
+      .status(400)
+      .json({ success: false, msg: 'please provide name value' })
+  }
+  res.status(201).json({ success: true, person: name })
+})
+app.post('/api/postman/people', (req, res) => {
+  const { name } = req.body
+  if (!name) {
+    return res
+      .status(400)
+      .json({ success: false, msg: 'please provide name value' })
+  }
+  res.status(201).json({ success: true, data: [...people, name] })
+})
+
+app.post('/login', (req, res) => {
+  const { name } = req.body
+  if (name) {
+    return res.status(200).send(`Welcome ${name}`)
+  }
+  res.status(401).send('Please Provide Credentials')
+})
+
+//PUT request
+app.put('/api/people/:id', (req, res) => {
```

Comment:
I believe it would be more organized if moved to the paople's controller/router.